test/reductions/grid_search/test_grid_search_arguments.py

```diff
@@ -0,0 +1,290 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import logging
+import numpy as np
+import pandas as pd
+import pytest
+from sklearn.linear_model import LogisticRegression, LinearRegression
+
+from fairlearn.metrics import DemographicParity, BoundedGroupLoss
+from fairlearn.reductions import GridSearch
+from fairlearn.reductions.grid_search.simple_quality_metrics import SimpleClassificationQualityMetric, SimpleRegressionQualityMetric  # noqa: E501
+
+# ==============================================================
+
+# The following are functions which convert ndarrays into other datatypes
+# They are used to generate different argument types for calls to
+# GridSearch
+
+
+def identity(X):
+    return X
+
+
+def pandasdf(X):
+    return pd.DataFrame(X)
+
+
+def pandasseries(X):
+    # Will not work if ndarray has more than one dimension
+    return pd.Series(X)
+
+
+def ndarray2d(X):
+    # Adds a second dimension of length 1 onto a 1D
+    # array. This is for checking that shapes n and
+    # n*1 behave the same
+    if len(X.shape) != 1:
+        raise RuntimeError("ndarray2d requires 1d ndarray")
+
+    X.shape = (X.shape[0], 1)
+    return X
+
+
+# List the different datatypes which need to succeed for
+# all GridSearch calls
+Xtransform = [identity, pandasdf]
+Ytransform = [identity, pandasdf, pandasseries, ndarray2d]
+Atransform = [identity, pandasdf, pandasseries, ndarray2d]
+
+
+# Base class for tests
+# Tests which must be passed by all calls to the GridSearch
+# go here
+class ArgumentTests:
+    def _quick_data(self, number_samples=48):
+        feature_1 = np.random.randint(2, size=number_samples)
+        feature_2 = np.random.randint(6, size=number_samples)
+        feature_3 = np.random.randint(3, size=number_samples)
+        X = np.stack((feature_1, feature_2, feature_3), -1)
+        Y = np.random.randint(2, size=number_samples)
+        A = np.random.randint(2, size=number_samples)
+        return X, Y, A
+
+    # ----------------------------
+
+    @pytest.mark.parametrize("transformA", Atransform)
```

Comment:
nit: this looks a little funny since it's transformA once and Atransform the other time.

Reply:
Which 'other time' ? I'm confused.

Reply:
same line :-)

Reply:
Well, once it's the name of the argument to a method, the other time it's the list of arguments to use. I suppose I could have 'transform_list_for_A` or something like that?

Reply:
Sure, but the inverted naming doesn't express that either :-)
Like I wrote above it's a "nit".... just thought it's odd.

Reply:
Have made the latter more descriptive